### DISCUSSION TITLE
Comments Plugin - Ability to Pass State Value to Automation That Uses Row Updated Trigger

### DISCUSSION THREAD

**Discussion by chadgriff** [Budibase/budibase#11520] [Help] [answered]
I am using the comments plugin to capture comments associated with a record. However, the plugin immediately updates the table when either the Enter key is selected or the Post button is clicked.  I would like to setup an automation to send out a notification when a new comment is added. I created an automation that uses the Row Updated trigger.  This works just fine.  However, I want to have logic in the automation so that it will only send out notification when a user assigned to a certain role is logged in.  My thought was to set a State Value upon page load `{{State:CurrentUserRole}}` of the current user's role.  However, I cannot seem to figure out how to get the Condition step within the automation to "see" the State value to compare it to the comparison value. It doesn't resolve the block but rather just sees the literal string "{{State.CurrentUserRole}}".  Any ideas of how I could make this work or achieve this desired use case?
![image](https://github.com/Budibase/budibase/assets/15015476/0670b156-3c8d-4085-ba31-c604684534fd)

**Comment by melohagan** [comment, marked as answer]
Hey @chadgriff 

Issues specific to a plugin are best asked on that repo 😄 

With that said, app state cannot be accessed in automations as they are server-side, and app state only works client-side.

You could pass the app state through to an automation by using the **App action** trigger: https://docs.budibase.com/docs/app-action

**Reply by chadgriff** [reply]
Ahhh...apologies. I have no issues with the plugin itself...it appears to be working exactly as described.  Rather, my question centers more on automations and passing values to an automation that uses a "Record Updated" trigger (i.e. I can't manually setup an input value like when "App Action" trigger is used.

**Reply by chadgriff** [reply]
Responded to quickly...thanks for the clarification / confirmation!

**Reply by melohagan** [reply]
Got it - thanks for clarifying! 

If you must use the **Row Updated** trigger, then you could try saving the app state to a table, which could then be queried by a **Query Rows** action. Just an idea.

**Reply by chadgriff** [reply]
Hmm...interesting approach that just might work. Thanks for the idea
